src/krate.rs

```diff
@@ -547,13 +549,28 @@ impl Crate {
         }
     }
 
+    // Return None if the documentation URL host matches a blacklisted host
     fn remove_blacklisted_documentation_urls(url: Option<String>) -> Option<String> {
-        let blacklisted_url = "rust-ci.org";
+        let url = match url {
+            Some(url) => url,
+            None => return None,
+        };
+
+        // Handles the case if the documentation URL does not parse successfully
+        let parsed_url = match Url::parse(&url) {
+            Ok(parsed_url) => parsed_url,
+            Err(_) => return None,
+        };
```

Comment:
I've been trying to see if we can use methods like `map` on the `Result`s and `Option`s in here, but it's just complex enough that I don't think the end result is any clearer than what you have here. I see what you mean about handling lots of cases!! (again nothing to do here I'm just talking)